x/emissions/keeper/queryserver/query_server_topics.go

```diff
@@ -89,3 +89,141 @@ func (qs queryServer) GetTopicLastReputerCommitInfo(ctx context.Context, req *ty
 
 	return &types.QueryTopicLastCommitResponse{LastCommit: &lastCommit}, nil
 }
+
+func (qs queryServer) GetTopicRewardNonce(
+	ctx context.Context,
+	req *types.QueryTopicRewardNonceRequest,
+) (
+	*types.QueryTopicRewardNonceResponse,
+	error,
+) {
+	nonce, err := qs.k.GetTopicRewardNonce(ctx, req.TopicId)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &types.QueryTopicRewardNonceResponse{Nonce: nonce}, nil
+}
+
+func (qs queryServer) GetPreviousTopicWeight(
+	ctx context.Context,
+	req *types.QueryPreviousTopicWeightRequest,
+) (
+	*types.QueryPreviousTopicWeightResponse,
+	error,
+) {
+	previousTopicWeight, notFound, err := qs.k.GetPreviousTopicWeight(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryPreviousTopicWeightResponse{Weight: previousTopicWeight, NotFound: notFound}, nil
+}
+
+func (qs queryServer) TopicExists(
+	ctx context.Context,
+	req *types.QueryTopicExistsRequest,
+) (
+	*types.QueryTopicExistsResponse,
+	error,
+) {
+	exists, err := qs.k.TopicExists(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryTopicExistsResponse{Exists: exists}, nil
+}
+
+func (qs queryServer) IsTopicActive(
+	ctx context.Context,
+	req *types.QueryIsTopicActiveRequest,
+) (
+	*types.QueryIsTopicActiveResponse,
+	error,
+) {
+	isActive, err := qs.k.IsTopicActive(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryIsTopicActiveResponse{IsActive: isActive}, nil
+}
+
+func (qs queryServer) GetIdsOfActiveTopics(
```

Comment:
this query already exists. can remove

Reply:
I'm not seeing where it's already defined, can you show me?

Reply:
found it, we have `GetActiveTopics` but not `GetIdsOfActiveTopics`. That works. I will delete.